src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -48,24 +50,43 @@ private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Pre
      */
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args,
-                PREFIX_NAME, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_ROLE, PREFIX_TIMEZONE,
-                PREFIX_PHONE, PREFIX_EMAIL, PREFIX_SLACK, PREFIX_TELEGRAM);
+            PREFIX_NAME, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_ROLE, PREFIX_TIMEZONE,
+            PREFIX_PHONE, PREFIX_EMAIL, PREFIX_SLACK, PREFIX_TELEGRAM, PREFIX_GITHUB);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)
+        if (!(arePrefixesPresent(argMultimap, PREFIX_NAME) || arePrefixesPresent(argMultimap, PREFIX_GITHUB))
             || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
-        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        // At least one of the following needs to be non-null
+        User githubUser = null;
+        Name name;
+
+        if (argMultimap.getValue(PREFIX_GITHUB).isPresent()) {
+            githubUser = ParserUtil.parseGithubUser(argMultimap.getValue(PREFIX_GITHUB).get());
+        }
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        } else {
+            assert githubUser != null;
+            name = githubUser.getName();
```

Comment:
I think either way works, I am not opposed to moving it to `Person` but I think it being here in the parser is fine as well cause we are just choosing which term to initialise name with. Moving it might cause further errors too as now we need to allow `null` names for `Person` constructor.

Reply:
Putting at here means that the logic will be duplicated in other place else. Just to imagine we have two AddCommand, and we need to implement this logic in two places, which is not "defensive" enough.